pkg/controller/command/outofband/command.go

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package outofband
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/hyperledger/aries-framework-go/pkg/client/outofband"
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/controller/command"
+	"github.com/hyperledger/aries-framework-go/pkg/controller/internal/cmdutil"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/internal/logutil"
+)
+
+const (
+	// InvalidRequestErrorCode is typically a code for validation errors
+	// for invalid outofband controller requests
+	InvalidRequestErrorCode = command.Code(iota + command.Outofband)
+	// CreateRequestErrorCode is for failures in create request command
+	CreateRequestErrorCode
+	// CreateInvitationErrorCode is for failures in create invitation command
+	CreateInvitationErrorCode
+	// AcceptRequestErrorCode is for failures in accept request command
+	AcceptRequestErrorCode
+	// AcceptInvitationErrorCode is for failures in accept invitation command
+	AcceptInvitationErrorCode
+)
+
+const (
+	// command name
+	commandName      = "outofband"
+	createRequest    = "CreateRequest"
+	createInvitation = "CreateInvitation"
+	acceptRequest    = "AcceptRequest"
+	acceptInvitation = "AcceptInvitation"
+
+	// error messages
+	errOneAttachmentMustBeProvided = "at least one attachment must be provided"
+	errEmptyRequest                = "request was not provided"
+	errEmptyMyLabel                = "my_label was not provided"
+	// log constants
+	successString = "success"
+)
+
+var logger = log.New("aries-framework/controller/outofband")
+
+// Command is controller command for outofband
+type Command struct {
+	client *outofband.Client
+}
+
+// New returns new outofband controller command instance
+func New(ctx outofband.Provider) (*Command, error) {
+	client, err := outofband.New(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create a client: %w", err)
+	}
+
+	// creates action channel
+	actions := make(chan service.DIDCommAction)
+	// registers action channel to listen for events
+	if err := client.RegisterActionEvent(actions); err != nil {
+		return nil, fmt.Errorf("register action event: %w", err)
+	}
+
+	// this code listens for the action events and does nothing
+	// this trick is used to avoid error such as `no clients are registered to handle the message`
+	go func() {
+		for range actions {
+		}
```

Comment:
We need to notify the consumers(rest/wasm) when we receive the events. Is it planned for future PRs ?

Reply:
I think it is a common problem. We do not have that mechanism for now. We have something custom for didexchange(notifier) but I don't want to repeat that. There should be a common logic for all protocols. Since all protocols are using the same common structures for the `DIDCommAction` and `StateMsg`.  I have not planned it yet. But I know about it.

Reply:
Ok. Just wanted to check as it's an essential components for the controller (wasm/rest controllers need to listen to events rather than polling for events).

Reply:
@soluchok please create a followup issue for this

Reply:
Already did) https://github.com/hyperledger/aries-framework-go/issues/1887